catalog_reader/questions.py

```diff
@@ -35,23 +35,74 @@ def normalize_question(question: dict, version_data: dict, context: dict) -> Non
             data['enum'] = [
                 {'value': i, 'description': f'{i!r} Interface'} for i in context['nic_choices']
             ]
-        elif ref == 'definitions/gpuConfiguration':
+        elif ref == 'definitions/gpu_configuration':
+            # How we will go about this is the following:
+            # If user has only nvidia gpus and we are able to retrieve nvidia gpu's uuid
+            # we will allow user to select any of the available nvidia gpu
+            # If user has any other GPU apart from nvidia vendor, he will have to passthrough all available gpus
+            # If user has nvidia + other gpu's, then we can show a boolean or a string enum
+            # highlighting the nvidia bits
+            gpu_choices = [entry for entry in context['gpu_choices'] if entry['gpu_details']['available_to_host']]
+            show_all_gpus_flag = any(
+                g['vendor'] != 'NVIDIA' or not g['vendor_specific_config'].get('uuid') for g in gpu_choices
+            )
+            show_nvidia_selection = any(
+                g['vendor'] == 'NVIDIA' and g['vendor_specific_config'].get('uuid') for g in gpu_choices
+            )
             data['attrs'] = [
                 {
-                    'variable': gpu,
-                    'label': f'GPU Resource ({gpu})',
-                    'description': 'Please enter the number of GPUs to allocate',
+                    'variable': 'use_all_gpus',
+                    'label': 'Passthrough available GPUs',
+                    'description': 'Please select this option to passthrough all available GPUs to the app',
```

Comment:
I think we should rephrase this a bit.

Nvidia gpus can be passed independently of intel/amd.
So even if user selects `use_all_gpus` user can still pick a specific nvidia gpu to pass.

So proposing to rephrase
"Passthrough available (non-NVIDIA) GPUs"
"Please select this option to passthrough all available (non-NVIDIA) GPUs to the app"

Reply:
Unless we want to pass actually all gpus here (including nvidia).
Then we should add a `show_if: [["use_all_gpus", "=", false]]` on the `nvidia_gpu_selection`

Reply:
@stavros-k is it possible on a system having AMD/INTEL and NVIDIA GPUs, and you are only passing through AMD/INTEL gpus ?

Reply:
Yes, amd/intel are passed via `/dev/dri` while nvidia via complete other definition

Reply:
@stavros-k so the intended behavior now is that if the boolean for use_all_gpus is set, all non-nvidia gpu's will be passed through

Reply:
Correct! Thanks